File: Matriz5x5.java
import java.util.Scanner;
import java.util.ArrayList;

public class Matriz5x5 {
    public static void main(String[] args) {
        Scanner leer = new Scanner(System.in);
        int[][] matriz = new int[5][5];
        
        // 1. Ingreso de datos
        System.out.println("INGRESO DE DATOS Matriz 5x5");
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                System.out.print("Ingrese valor para [" + i + "][" + j + "]: ");
                matriz[i][j] = leer.nextInt();
            }
        }

        // 2. Mostrar matriz en forma de tabla
        System.out.println("\nMatriz ingresada:");
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }

        // 3.Aplanado usando ArrayList
        System.out.println("--------------------");
        ArrayList<Integer> listaAplanada = new ArrayList<>();
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                listaAplanada.add(matriz[i][j]);
            }
        }
        System.out.println("Matriz aplanada (lista): " + listaAplanada);
    }
}

File: matriz5x5.py
# 1. Crear matriz 5x5 empezando en 0
matriz = [[0 for _ in range(5)] for _ in range(5)]

# 2. Ingreso los datos
print("--- Ingreso de datos para la Matriz 5x5 (Python) ---")
for i in range(5):
    for j in range(5):
        valor = int(input(f"Ingrese valor para [{i}][{j}]: "))
        matriz[i][j] = valor

# 3. Mostrar la matriz en forma de tabla
print("\nMatriz ingresada:")
for i in range(5):
    for j in range(5):
        print(matriz[i][j], end="\t")
    print() # Salto de línea

# 4. Aplanado
print("-" * 20)
lista_aplanada = []
for fila in matriz:
    for columna in fila:
        lista_aplanada.append(columna)
print("Matriz aplanada (lista):", lista_aplanada)